src/test/java/com/cdancy/bitbucket/rest/features/BranchApiMockTest.java

```diff
@@ -100,6 +114,33 @@ public void testListBranches() throws Exception {
         }
     }
 
+
+    private static class BuildStatusInfo {
```

Comment:
@vasilievip I wouldn't call passing around a `JsonElement` as being anything "internal". It's simply an admission that the json returned can't be modeled for all use-cases.

Yeah I've looked at `feign` previously and the original author of that library is the original author of `jclouds` which this client is based on. At the time of starting this library they didn't have jax-rs support, which is what I liked about jclouds, but I believe they've provided a hook of sorts within the last year or so to make this possible. I like the opinionated and focused view of `jclouds` whereas with `feign` it seems to want to support any and everything.